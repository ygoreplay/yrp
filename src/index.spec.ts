import { Replay } from ".";

import * as fs from "fs";
import path from "path";

describe("yrp", () => {
    it("should uncompress replay header information properly", async () => {
        const originalFileBuffer = fs.readFileSync(path.join(process.cwd(), "./res/yrp-basic.yrp"));
        const replay = await Replay.fromBuffer(originalFileBuffer);

        expect(replay.getReplayHeader()).toMatchSnapshot();
    });

    it("should uncompress replay header information from file properly", async () => {
        const replay = await Replay.fromFile(path.join(process.cwd(), "./res/yrp-basic.yrp"));
        expect(replay.getReplayHeader()).toMatchSnapshot();
    });
});
