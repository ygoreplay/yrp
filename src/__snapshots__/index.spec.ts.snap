// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yrp should uncompress replay header information from file properly 1`] = `
Object {
  "dataSize": 442,
  "flag": 1,
  "hash": 0,
  "id": 829452921,
  "props": Object {
    "data": Array [
      93,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
    ],
    "type": "Buffer",
  },
  "seed": 1616370681,
  "version": 4946,
}
`;

exports[`yrp should uncompress replay header information properly 1`] = `
Object {
  "dataSize": 442,
  "flag": 1,
  "hash": 0,
  "id": 829452921,
  "props": Object {
    "data": Array [
      93,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
    ],
    "type": "Buffer",
  },
  "seed": 1616370681,
  "version": 4946,
}
`;
